vsphere/resource_vsphere_tag_category_test.go

```diff
@@ -0,0 +1,308 @@
+						),
+					},
+					{
+						Config:      testAccResourceVSphereTagCategoryConfigBasic,
+						ExpectError: regexp.MustCompile("removal of associable types is not supported"),
+					},
+				},
+			},
+		},
+		{
+			"rename",
+			resource.TestCase{
+				PreCheck: func() {
+					testAccPreCheck(tp)
+				},
+				Providers:    testAccProviders,
+				CheckDestroy: testAccResourceVSphereTagCategoryExists(false),
+				Steps: []resource.TestStep{
+					{
+						Config: testAccResourceVSphereTagCategoryConfigBasic,
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereTagCategoryExists(true),
+						),
+					},
+					{
+						Config: testAccResourceVSphereTagCategoryConfigAltName,
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereTagCategoryExists(true),
+							testAccResourceVSphereTagCategoryHasName("terraform-test-category-renamed"),
+						),
+					},
+				},
+			},
+		},
+		{
+			"single cardinality",
+			resource.TestCase{
+				PreCheck: func() {
+					testAccPreCheck(tp)
+				},
+				Providers:    testAccProviders,
+				CheckDestroy: testAccResourceVSphereTagCategoryExists(false),
+				Steps: []resource.TestStep{
+					{
+						Config: testAccResourceVSphereTagCategoryConfigSingleCardinality,
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereTagCategoryExists(true),
+							testAccResourceVSphereTagCategoryHasCardinality(vSphereTagCategoryCardinalitySingle),
+						),
+					},
+				},
+			},
+		},
+		{
+			"import",
+			resource.TestCase{
+				PreCheck: func() {
+					testAccPreCheck(tp)
+				},
+				Providers:    testAccProviders,
+				CheckDestroy: testAccResourceVSphereTagCategoryExists(false),
+				Steps: []resource.TestStep{
+					{
+						Config: testAccResourceVSphereTagCategoryConfigBasic,
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereTagCategoryExists(true),
+						),
+					},
+					{
+						ResourceName:      "vsphere_tag_category.terraform-test-category",
+						ImportState:       true,
+						ImportStateVerify: true,
+						ImportStateIdFunc: func(s *terraform.State) (string, error) {
+							cat, err := testGetTagCategory(s, "terraform-test-category")
+							if err != nil {
+								return "", err
+							}
+							return cat.Name, nil
+						},
+						Config: testAccResourceVSphereTagCategoryConfigBasic,
+						Check: resource.ComposeTestCheckFunc(
+							testAccResourceVSphereTagCategoryExists(true),
+						),
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testAccResourceVSphereTagCategoryCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tp = t
+			resource.Test(t, tc.testCase)
+		})
+	}
+}
+
+func testAccResourceVSphereTagCategoryExists(expected bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		_, err := testGetTagCategory(s, "terraform-test-category")
+		if err != nil {
+			if strings.Contains(err.Error(), "Status code: 404") && !expected {
```

Comment:
is there no way of getting a concrete object back and checking the status code on that instead?

Reply:
Nope. The library does not wrap an error type to pull it from ([reference](https://github.com/vmware/vic/blob/master/pkg/vsphere/tags/categories.go#L120)).